manifests/enterprise/config.pp

```diff
@@ -0,0 +1,149 @@
+# Class splunk::enterprise::config
+#
+class splunk::enterprise::config(
+){
+
+  if $splunk::enterprise::manage_password {
+    file { $splunk::enterprise::password_config_file:
+      ensure  => file,
+      owner   => $splunk::enterprise::splunk_user,
+      group   => $splunk::enterprise::splunk_user,
+      content => $splunk::enterprise::password_content,
+    }
+
+    file { $splunk::enterprise::secret_file:
+      ensure  => file,
+      owner   => $splunk::enterprise::splunk_user,
+      group   => $splunk::enterprise::splunk_user,
+      content => $splunk::enterprise::secret,
+    }
+  }
+
+  # Remove init.d file if the service provider is systemd
+  if $facts['service_provider'] == 'systemd' and versioncmp($splunk::enterprise::version, '7.2.2') >= 0 {
+    file { '/etc/init.d/splunk':
+      ensure => 'absent',
+    }
+  }
+
+  if $facts['virtual'] == 'docker' {
+    ini_setting { 'OPTIMISTIC_ABOUT_FILE_LOCKING':
```

Comment:
lol, nice name for an option :D

Reply:
Lol, most of this is reused code I didn't find reason to modify :)